sideboard/lib/_profiler.py

```diff
@@ -0,0 +1,271 @@
+    """
+    filename, line, name = func_name
+    if filename.endswith('__init__.py'):
+        return os.path.basename(filename[:-12]) + filename[-12:], line, name
+    return os.path.basename(filename), line, name
+
+pstats.func_strip_path = _new_func_strip_path
+
+
+@entry_point
+def cleanup_profiler():
+    """
+    Deletes all `*.prof` files in the profiler's data directory.
+
+    This is useful when you've created tons of profile files that you're no
+    longer interested in. Exposed as a `sep` command::
+
+        $ sep cleanup_profiler
+
+    The profiler directory is specified in the config by::
+
+        [cherrypy]
+        profiling.path = 'path/to/profile/data'
+
+    """
+    profiling_path = config['cherrypy'].get('profiling.path', None)
+    if profiling_path:
+        for f in glob(os.path.join(profiling_path, '*.prof')):
+            os.remove(f)
+
+
+def profile(func):
+    """
+    Decorator to capture profile data from a method or function.
+
+    If profiling is disabled then this decorator is a no-op, and the original
+    function is returned unmodified. Since the original function is returned,
+    this decorator does not incur any performance penalty if profiling is
+    disabled. To enable or disable profiling use the following setting in your
+    config::
+
+            [cherrypy]
+            profiling.on = 'True'  # Or 'False' to disable
+
+    Args:
+        func (function): The function to profile.
+
+    Returns:
+        function: Either a wrapped version of `func` with profiling enabled,
+            or `func` itself if profiling is disabled.
+
+    See Also:
+        configspec.ini
+    """
+    if config['cherrypy'].get('profiling.on', False):
+        profiling_path = config['cherrypy'].get('profiling.path', None)
+        if config['cherrypy'].get('profiling.aggregate', False):
+            p = ProfileAggregator(profiling_path)
+        else:
+            p = Profiler(profiling_path)
+
+        @wraps(func)
+        def wrapper(*args, **kwargs):
+            return p.run(func, *args, **kwargs)
+        return wrapper
+    else:
+        return func
+
+
+class Profiler(object):
+    """
+    Mostly copied from cherrypy/lib/profiler.py.
+
+     * Adds the ability to sort results by different columns.
+     * Adds the ability to cleanup profile data files.
+     * Uses a better naming scheme for profile data files.
+    """
+
+    # https://docs.python.org/3/library/profile.html#pstats.Stats.sort_stats
+    sort_fields = [
+        ('cumulative', 'Cumulative Time'),
+        ('filename', 'File Name'),
+        ('ncalls', 'Call Count'),
+        ('pcalls', 'Primitive Call Count'),
+        ('line', 'Line Number'),
+        ('name', 'Function Name'),
+        ('nfl', 'Function/File/Line'),
+        ('stdname', 'Standard Name'),
+        ('tottime', 'Total Time')]
+
+    def __init__(self, path=None):
+        if not path:
+            path = os.path.join(os.path.dirname(__file__), 'profile')
```

Comment:
This seems like a bad default, since this makes the default inside the ``sideboard/lib`` directory.  Unless I'm misunderstanding, I think it would be better to just say this:

```python
def __init__(self, path=config['cherrypy']['profiling.path']):
```

I realize that in practice we're always passing a value, but leaving a nonsensical default value here seems bad.